app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/attachment/AttachmentOptions.kt

```diff
@@ -9,27 +9,39 @@ import androidx.compose.foundation.layout.PaddingValues
 import androidx.compose.foundation.layout.fillMaxWidth
 import androidx.compose.foundation.lazy.GridCells
 import androidx.compose.foundation.lazy.LazyVerticalGrid
+import androidx.compose.material3.ExperimentalMaterial3Api
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import com.wire.android.R
 import com.wire.android.appLogger
 import com.wire.android.ui.common.AttachmentButton
 import com.wire.android.ui.common.dimensions
+import com.wire.android.ui.home.conversations.model.AttachmentBundle
+import com.wire.android.ui.home.messagecomposer.AttachmentInnerState
+import com.wire.android.ui.home.messagecomposer.AttachmentState
 import com.wire.android.util.permission.UseCameraRequestFlow
+import com.wire.android.util.permission.UseStorageRequestFlow
 import com.wire.android.util.permission.rememberCaptureVideoFlow
 import com.wire.android.util.permission.rememberCurrentLocationFlow
 import com.wire.android.util.permission.rememberOpenFileBrowserFlow
 import com.wire.android.util.permission.rememberOpenGalleryFlow
 import com.wire.android.util.permission.rememberRecordAudioRequestFlow
 import com.wire.android.util.permission.rememberTakePictureFlow
 
-@OptIn(ExperimentalComposeUiApi::class, ExperimentalFoundationApi::class)
+@OptIn(ExperimentalComposeUiApi::class, ExperimentalFoundationApi::class, ExperimentalMaterial3Api::class)
 @Composable
-fun AttachmentOptionsComponent() {
-    val attachmentOptions = buildAttachmentOptionItems()
+fun AttachmentOptionsComponent(
+    attachmentInnerState: AttachmentInnerState,
+    onSendAttachment: (AttachmentBundle?) -> Unit,
+    onError: (String) -> Unit
+) {
+    val attachmentOptions = buildAttachmentOptionItems(attachmentInnerState)
```

Comment:
how about instead of passing attachmentInnerState to buildAttachmentOptionItems we pass an lambda like : filePicked : (Uri) -> {} and call buildAttachmentOptionItems like buildAttachmentOptionItems{ onFilePicked = 
uri -> attachmentInnerState.pickAttachment(context, pickedPictureUri }   ? that way it is clear while you are reading what you want to do with the the picked file,  you do not have to jump to buildAttachmentOptionItems first and then to FileBrowserFlow to see what we are going to do with that  attachmentInnerState, let me know what you think

Reply:
I think indeed will help in not passing around or getting the context each time for every sharing flow.
This changes with the other from below on the Inner state, makes life easier when calling the function 👍